realtimenet/controller.py

```diff
@@ -0,0 +1,174 @@
+from collections import Callable
+from typing import List
+from typing import Optional
+from typing import Union
+
+from realtimenet.camera import VideoSource
+from realtimenet.camera import VideoStream
+from realtimenet.display import DisplayResults
+from realtimenet.engine import InferenceEngine
+from realtimenet.downstream_tasks.postprocess import PostProcessor
+
+import cv2
+import numpy as np
+import torch.nn as nn
+
+
+class Controller:
+    def __init__(
+            self,
+            neural_network: nn.Module,
+            post_processors: Union[PostProcessor, List[PostProcessor]],
+            results_display: DisplayResults,
+            callbacks: Optional[List[Callable]] = None,
+            camera_id: int = 0,
+            path_in: str = Optional[None],
+            path_out: str = Optional[None],
+            use_gpu: bool = True):
+        """
+        :param neural_network:
+            The neural network that produces the predictions for the camera image.
+            TODO: Be more specific here
```

Comment:
Remaining TODO. On a related note, `neural_network: nn.Module` might be a bit misleading because not every `nn.Module` will work as `InferenceEngine` expects `net` to define certain attrivutes/methods (`net.fps`, `net.step_size`, `net.preprocess` etc...).  We should create an abstract RealtimeNeuralNet class that clarifies the requirements. Should be done in another PR; let's leave the TODO here for now.